Project2015To2017.Core/Definition/ProjectExtensions.cs

```diff
@@ -45,11 +51,39 @@ public static (IReadOnlyList<XElement> unconditional, IReadOnlyList<(string cond
 
 		public static XElement Property(this Project project, string name, bool tryConditional = false)
 		{
+			if (name == null) throw new ArgumentNullException(nameof(name));
+
 			var (unconditional, conditional) = project.PropertyAll(name);
 			return unconditional.LastOrDefault()
 			       ?? (tryConditional ? conditional.Select(x => x.element).LastOrDefault() : null);
 		}
 
+		public static IEnumerable<(Guid guid, XElement source)> IterateProjectTypeGuids(this Project project)
+		{
+			var (unconditional, conditional) = project.PropertyAll("ProjectTypeGuids");
+			if (conditional.Count > 0)
+			{
+				// log unexpected case
+			}
+
+			var guidTypes = new HashSet<Guid>();
+			foreach (var element in unconditional)
+			{
+				// parse the CSV list
+				foreach (var guid in element.Value
+					.Split(';')
+					.Select(x =>
+						Guid.TryParse(x.Trim().ToUpperInvariant(), out var res)
```

Comment:
That toupperinvariant doesn't really matter here, right?
```suggestion
						Guid.TryParse(x.Trim(), out var res)
```

Reply:
Well, it's not specified in docs, but implementations actually don't care about casing, so yeah, I've fixed that.